packages/autocomplete-plugin-tags/src/createTagsPlugin.tsx

```diff
@@ -0,0 +1,173 @@
+/** @jsx createElement */
+import { BaseItem, PluginSubscribeParams } from '@algolia/autocomplete-core';
+import {
+  AutocompletePlugin,
+  AutocompleteSource,
+  AutocompleteState,
+} from '@algolia/autocomplete-js';
+
+import { createTags, OnTagsChangeParams } from './createTags';
+import type { BaseTag, Tag } from './types';
+
+type OnChangeParams<TTag> = PluginSubscribeParams<any> &
+  OnTagsChangeParams<TTag>;
+
+type TagsPluginData<TTag> = {
+  /**
+   * Returns the current list of tags.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-tags
+   */
+  readonly tags: Array<Tag<TTag>>;
+  /**
+   * Adds tags to the list.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-addtags
+   */
+  addTags: (tags: Array<BaseTag<TTag>>) => void;
+  /**
+   * Sets the list of tags.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-settags
+   */
+  setTags: (tags: Array<BaseTag<TTag>>) => void;
+};
+
+export type CreateTagsPluginParams<
+  TTag extends Record<string, any>,
+  TItem extends BaseItem
+> = {
+  /**
+   * A set of initial tags to pass to the plugin.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-initialtags
+   */
+  initialTags?: Array<BaseTag<TTag>>;
+  /**
+   * A function to specify what sources the plugin should subscribe to. The plugin adds a tag when selecting an item from these sources.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-gettagssubscribers
+   */
+  getTagsSubscribers?(): Array<{
+    sourceId: string;
+    getTag(params: { item: TItem }): BaseTag<TTag>;
+  }>;
+  /**
+   * A function to transform the returned tags source.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-transformsource
+   */
+  transformSource?(params: {
+    source: AutocompleteSource<Tag<TTag>>;
+    state: AutocompleteState<Tag<TTag>>;
+  }): AutocompleteSource<Tag<TTag>> | undefined;
+  /**
+   * A function called when the list of tags changes.
+   *
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-tags/createtagsplugin/#param-onchange
+   */
+  onChange?(params: OnChangeParams<TTag>): void;
+};
+
+export function createTagsPlugin<TTag, TItem extends BaseItem>({
+  initialTags = [],
+  getTagsSubscribers = () => [],
+  transformSource = ({ source }) => source,
+  onChange = () => {},
```

Comment:
We should be able to use our internal `noop`.

Reply:
Yup, I'll move it from `core` to `shared` then.

Reply:
Handled in 55a929d.